json-schema/json-schema/src/main/scala/endpoints4s/algebra/JsonSchemas.scala

```diff
@@ -238,6 +238,24 @@ trait JsonSchemas extends TuplesSchemas with PartialInvariantFunctorSyntax {
     */
   def lazyTagged[A](schema: => Tagged[A], name: String): JsonSchema[A]
 
+  /** A lazy JSON schema that can references schemas currently being defined:
+    *
+    * {{{
+    *   case class Recursive(next: Option[Recursive])
+    *   val recursiveSchema: JsonSchema[Recursive] = lazySchema("Rec")(
+    *     optField("next")(recursiveSchema)
+    *   ).xmap(Recursive)(_.next)
+    * }}}
+    *
+    * Interpreters should return a JsonSchema value that does not evaluate
+    * the given `schema` unless it is effectively used.
+    *
+    * @param name A unique name identifying the schema
+    * @param schema The record JSON schema whose evaluation should be delayed
+    * @group operations
+    */
+  def lazySchema[A](name: String)(schema: => JsonSchema[A]): JsonSchema[A]
```

Comment:
In order to stay backward binary compatible, we need to provide a default implementation (which throws an error):

```suggestion
  def lazySchema[A](name: String)(schema: => JsonSchema[A]): JsonSchema[A] =
    sys.error(s"Unsupported algebra version: 1.4.0. Please update your interpreter.")
```

Reply:
Also, are there any benefits in keeping `lazyRecord` and `lazyTagged`? Should we just deprecate them in favor of `lazySchema`? If we keep them, would it make sense to add two overloads with signatures looking similar to `lazySchema`?

Reply:
I marked `lazyRecord` and `lazyTagged` as deprecated. The migration is indeed pretty straight forward.